posthog/tasks/status_report.py

```diff
@@ -63,31 +69,57 @@ def status_report(*, dry_run: bool = False) -> Dict[str, Any]:
 
     for team in Team.objects.exclude(organization__for_internal_metrics=True):
         try:
+            params = (team.id, report["period"]["start_inclusive"], report["period"]["end_inclusive"])
             team_report: Dict[str, Any] = {}
-            events_considered_total = Event.objects.filter(team_id=team.id)
-            instance_usage_summary["events_count_total"] += events_considered_total.count()
-            events_considered_new_in_period = events_considered_total.filter(
-                timestamp__gte=period_start, timestamp__lte=period_end,
-            )
+            if is_clickhouse_enabled():
+                # pull events stats from clickhouse
+                from ee.clickhouse.models.event import (
+                    get_event_count_for_team,
+                    get_event_count_for_team_and_period,
+                    get_events_count_for_team_by_client_lib,
+                    get_events_count_for_team_by_event_type,
+                )
+
+                team_event_count = get_event_count_for_team(team.id)
+                instance_usage_summary["events_count_total"] += team_event_count
+                team_report["events_count_total"] = team_event_count
+                team_events_in_period_count = get_event_count_for_team_and_period(team.id, period_start, period_end)
+                team_report["events_count_new_in_period"] = team_events_in_period_count
+                instance_usage_summary["events_count_new_in_period"] += team_report["events_count_new_in_period"]
+
+                team_report["events_count_by_lib"] = get_events_count_for_team_by_client_lib(
+                    team.id, period_start, period_end
+                )
+                team_report["events_count_by_name"] = get_events_count_for_team_by_event_type(
+                    team.id, period_start, period_end
+                )
+            else:
+                # pull events stats from postgres
+                events_considered_total = Event.objects.filter(team_id=team.id)
+                instance_usage_summary["events_count_total"] += events_considered_total.count()
+                events_considered_new_in_period = events_considered_total.filter(
+                    timestamp__gte=period_start, timestamp__lte=period_end,
+                )
+
+                team_report["events_count_total"] = events_considered_total.count()
```

Comment:
n00b Q: do we run the query when we call `count()` if so we are doing `events_considered_total.count()` twice unnecessarily here, we could use a variable like you did for clickhouse part or just take it from the `team_report` for `instance_usage_summary` like we do below. Also a consistency nit.

Reply:
love it but not going to spend any time on Postgres code right now. Can't wait to 🔥 it all 😈